src/highdicom/sr/content.py

```diff
@@ -853,6 +855,181 @@ def __init__(
+                'content item with concept name "Source Series For '
+                'Segmentation".'
+            )
+
+        if len(source_image_items) > 0:
+            new_seq = ContentSequence(
+                vol_surface_items + source_image_items
+            )
+        else:
+            if len(source_series_items) > 1:
+                raise RuntimeError(
+                    'Sequence should contain at most one content item '
+                    'with concept name "Source Series For Segmentation". Found '
+                    f'{len(source_series_items)}.'
+                )
+            new_seq = ContentSequence(
+                vol_surface_items + source_series_items
+            )
+
+        new_seq.__class__ = cls
+        new_seq = cast(VolumeSurface, new_seq)
+        new_seq._graphic_type = vol_surface_items[0].graphic_type
+        return new_seq
+
+    @property
+    def graphic_type(self) -> GraphicTypeValues3D:
+        """highdicom.sr.GraphicTypeValues3D: Graphic type."""
+        return self._graphic_type
+
+    @property
+    def frame_of_reference_uid(self) -> UID:
+        """highdicom.UID: Frame of reference UID."""
+        return UID(self._graphic_data_items[0].frame_of_reference_uid)
+
+    def has_source_images(self) -> bool:
+        """Returns whether the object contains information about source images.
+
+        ReferencedSegment objects must either contain information about source
+        images or source series (and not both).
+
+        Returns
+        -------
+        bool:
+            True if the object contains information about source images. False
+            if the image contains information about the source series.
+
+        """
+        return len(self._lut[codes.DCM.SourceImageForSegmentation]) > 0
+
+    @property
+    def source_images_for_segmentation(
+        self
+    ) -> List[SourceImageForSegmentation]:
+        """List[highdicom.sr.SourceImageForSegmentation]:
+            Source images for the volume surface.
+        """
+        return self._lut[codes.DCM.SourceImageForSegmentation]
+
+    @property
+    def source_series_for_segmentation(
+        self
+    ) -> Optional[SourceSeriesForSegmentation]:
+        """Optional[highdicom.sr.SourceSeriesForSegmentation]:
+            Source series for the volume surface.
+        """
+        items = self._lut[codes.DCM.SourceSeriesForSegmentation]
+        if len(items) == 0:
+            return None
+        else:
+            return cast(SourceSeriesForSegmentation, items[0])
+
+    @property
+    def _graphic_data_items(self) -> List[Scoord3DContentItem]:
+        """List[highdicom.sr.Scoord3DContentItem]:
+            Graphic data elements that comprise the volume surface.
+        """
+        return cast(
+            List[Scoord3DContentItem],
+            self._lut[codes.DCM.VolumeSurface]
+        )
+
+    @property
+    def graphic_data(self) -> List[np.ndarray]:
```

Comment:
I am wondering whether this should be `Union[numpy.ndarray, List[numpy.ndarray]]` to a) match the constructor argument and b) provide a single set of coordinates in case of `ELLIPSOID`.

What do you think?

Reply:
Generally I try to avoid having dynamically-determined returned types. But I think here maybe it makes sense and does match the input signature for the constructor. Implemented in 9ec6836